ouroboros-network-testing/src/Ouroboros/Network/Testing/Data/Script.hs

```diff
@@ -60,22 +61,44 @@ arbitraryScriptOf :: Int -> Gen a -> Gen (Script a)
 arbitraryScriptOf maxSz a =
     sized $ \sz -> do
       n <- choose (1, max 1 (min maxSz sz))
-      (Script . NonEmpty.fromList) <$> vectorOf n a
+      Script . NonEmpty.fromList <$> vectorOf n a
 
 initScript :: MonadSTM m => Script a -> m (TVar m (Script a))
 initScript = newTVarIO
 
 stepScript :: MonadSTM m => TVar m (Script a) -> m a
-stepScript scriptVar = atomically (stepScriptSTM scriptVar)
+stepScript scriptVar = atomically (stepScriptPureSTM' scriptVar)
 
-stepScriptSTM :: MonadSTMTx stm => TVar_ stm (Script a) -> stm a
-stepScriptSTM scriptVar = do
+initScriptSTM :: MonadSTMTx stm
+              => Script a
+              -> stm (TVar_ stm (Script a))
+initScriptSTM = LazySTM.newTVar
+
+stepScriptPureSTM' :: MonadSTMTx stm => TVar_ stm (Script a) -> stm a
+stepScriptPureSTM' scriptVar = do
     Script (x :| xs) <- readTVar scriptVar
     case xs of
       []     -> return ()
       x':xs' -> writeTVar scriptVar (Script (x' :| xs'))
     return x
 
+stepScriptPureSTM :: MonadSTMTx stm => TVar_ stm (Script a) -> stm a
+stepScriptPureSTM scriptVar = do
+    Script (x :| xs) <- LazySTM.readTVar scriptVar
+    case xs of
+      []     -> return ()
+      x':xs' -> LazySTM.writeTVar scriptVar (Script (x' :| xs'))
+    return x
+
+stepScriptSTM :: MonadSTMTx stm
+               => TVar_ stm (Script (stm b)) -> stm b
+stepScriptSTM scriptVar = do
+  Script (script :| scripts) <- LazySTM.readTVar scriptVar
+  case scripts of
+    []     -> return ()
+    x':xs' -> LazySTM.writeTVar scriptVar (Script (x' :| xs'))
+  script
```

Comment:
```suggestion
stepScriptSTM = stepScriptSTM >=> id
```

I think I would avoid it after all and retain the original api

Reply:
It doesn't seem used in this commit (*Factor Signal, Script, Utils modules*) as well.

Reply:
Hmm good point, I'll look into it!

Reply:
What was your conclusion?

Reply:
My conclusion was that it was not used in this PR but used in another one ( I added it under a different name, in the respective PR)